grease-exe/src/Grease/Main.hs

```diff
@@ -1229,32 +1250,40 @@ simulateMacawCfgs ::
   IO Results
 simulateMacawCfgs la halloc macawCfgConfig archCtx simOpts setupHook addrOvs cfgs = do
   let fm = W4.FloatRealRepr
-  withSolverOnlineBackend (simSolver simOpts) fm globalNonceGenerator $ \bak -> do
-    results <- do
-      let nEntries = Map.size cfgs
-      iforM (Map.toList cfgs) $ \i (entry, MacawEntrypointCfgs entrypointCfgs mbCfgAddr) ->
-        analyzeEntrypoint la entry i nEntries $ do
-          entrypointCfgsSome <- do
-            let usrErr err = do
-                  let url = "https://galoisinc.github.io/grease/sexp-progs.html"
-                  userError la (PP.pretty err <> "\n" <> "For more information, see " <> url)
-            case checkMacawEntrypointCfgsSignatures archCtx entrypointCfgs of
-              -- See Note [Explicitly listed errors]
-              Left err@GME.BadArgs{} -> usrErr err
-              Left err@GME.BadRet{} -> usrErr err
-              Right ok -> pure ok
-          status <- simulateMacawCfg la bak fm halloc macawCfgConfig archCtx simOpts setupHook addrOvs mbCfgAddr entrypointCfgsSome
-          let result =
-                Batch
-                  { batchStatus = status
-                  , batchLoadOffset =
-                      fromMaybe 0 $
-                        MML.loadOffset $
-                          mcLoadOptions macawCfgConfig
-                  }
-          pure (entry, result)
+  withMaybeFile (simDumpCoverage simOpts) WriteMode $ \covHandle -> do
+    withMaybeFile (simDumpSectionMap simOpts) WriteMode $ \secMapHandle -> do
+      let mem = mcMemory macawCfgConfig
+      let secsJson = Load.dumpSections mem
+      Maybe.maybe (pure ()) (\hand -> Text.IO.hPutStrLn hand (renderJSON secsJson)) secMapHandle
```

Comment:
Given the `Maybe.maybe` here, it seems like `withMaybeFile` is perhaps overly complex for what this code does. Perhaps:
```haskell
case simDumpSectionMap simOpts of
  Nothing -> pure ()
  Just path -> do
    let mem = mcMemory macawCfgConfig
    let secsJson = Load.dumpSections mem
    withFile path WriteMode $ \hand -> Text.IO.hPutStrLn hand (renderJSON secsJson)
```
This could also go above the `withMaybeFile` for `covHandle`, since it doesn't need access to `covHandle`.